devops/girder/annotation_client/annotation_client/annotations.py

```diff
@@ -403,3 +414,44 @@ def getPropertyValuesForAnnotation(self, datasetId, annotationId):
                 annotationId=annotationId, datasetId=datasetId
             )
         )
+
+    def setPropertiesByConfigurationId(self, configurationId, propertyIdList):
+        """
+        Set the properties for a configuration
+        :param str configurationId: The id of the configuration
+        :param list propertyIdList: The list of property ids to set
+        """
+        metadata = json.dumps({"propertyIds": propertyIdList})
+
+        params = {
+            'metadata': metadata
+        }
+
+        return self.client.put(
+            PATHS["properties_by_configuration"].format(
+                configurationId=configurationId
+            ),
+            parameters=params
+        )
+
+    # Configurations
+
+    def getPropertiesByConfigurationId(self, configurationId):
+        """
+        Get the properties for a configuration
+        :param str configurationId: The id of the configuration
+        """
+        return self.client.get(
+            PATHS["properties_by_configuration"].format(
+                configurationId=configurationId
+            )
+        )
+
+    def getConfigurationsByDatasetId(self, datasetId):
+        """
+        Get the configurations for a dataset
+        :param str datasetId: The id of the dataset
+        """
+        return self.client.get(
+            PATHS["configurations_by_dataset"].format(datasetId=datasetId)
+        )
```

Comment:
These two functions don't return exactly what they say they return:
- The first one returns the whole configuration item (let's call it `item`). So you will have to do `item.meta.propertyIds` to get the property ids, which is not even the properties themself.
- The second one returns a list of dataset views, which contain a `configurationId` but not the configuration.

To make the things a little bit cleaner you can rename them:
- `getItemById` and rename `PATHS["properties_by_configuration"]` to `PATHS["item_by_id"]`
- `findDatasetViewsByDatasetId` and rename `PATHS["configurations_by_dataset"]` to `PATHS["dataset_views_by_dataset"]`